src/NUnitFramework/framework/Internal/Builders/DefaultSuiteBuilder.cs

```diff
@@ -176,7 +176,7 @@ private bool HasArguments(IFixtureBuilder attr)
             // Only TestFixtureAttribute can be used without arguments
             var temp = attr as TestFixtureAttribute;
 
-            return temp == null || temp.Arguments.Length > 0 || temp.TypeArgs.Length > 0;
+            return temp == null || (temp.Arguments != null && temp.Arguments.Length > 0) || (temp.TypeArgs != null && temp.TypeArgs.Length > 0);
         }
```

Comment:
I'm a huge fan of failing early. As a consumer of NUnit, not running NUnit's own tests, if I write `[TestFixture(null)]` and that isn't legal, I'd expect an error to be thrown in the constructor. 

Inserting a single `ArgumentNullException` precondition on `arguments` [here](https://github.com/nunit/nunit/blob/release/3.6/src/NUnitFramework/framework/Attributes/TestFixtureAttribute.cs#L55) would be all that's needed, right? Either that or `?? new object[0]`.